src/models/decorators.ts

```diff
@@ -66,10 +67,50 @@ export function isColumnNullable(targetClass: Function, propertyName: string): b
     return false;
 }
 
-export function getPrimaryKey(targetClass: Function): string | symbol | undefined {
+export function getPrimaryKeys(targetClass: Function): string[] {
     const metadata = metadataRegistry.get(targetClass);
     if (metadata) {
-        return metadata.primaryKey;
+        // Get all the keys of the metadata object that are primary keys
+        const keys = Object.keys(metadata.columns).filter((key) => metadata.columns[key].primary);
+
+        // Return the actual column name as it is stored in the database
+        return keys.map((key) => metadata.columns[key].name);
+    }
+    return [];
+}
```

Comment:
The code can be more verbose here by making variable names that have meaning. 
Also you have the ability to use `Object.entries` which should make it so you don't have to do a key lookup in `metadata.columns`. Chain some functions 🚋 
```js
    const primaryKeyColumnsNames = Object.entries(metadata.columns)
        .filter(([key, column]) => column.primary)
        .map(([key, column]) => column.name);

    return primaryKeyColumnsNames;
```